library/std/src/sys/random/vxworks.rs

```diff
@@ -0,0 +1,25 @@
+use crate::sync::atomic::AtomicBool;
+use crate::sync::atomic::Ordering::Relaxed;
+
+static RNG_INIT: AtomicBool = AtomicBool::new(false);
+
+pub fn fill_bytes(mut bytes: &mut [u8]) {
+    while !RNG_INIT.load(Relaxed) {
+        let ret = unsafe { libc::randSecure() };
+        if ret < 0 {
+            panic!("failed to generate random data");
+        } else if ret > 0 {
+            RNG_INIT.store(true, Relaxed);
+            break;
+        }
+
+        unsafe { libc::usleep(10) };
```

Comment:
Is there some documentation for why this is needed or desired? Could this get a comment, please?

Reply:
I guess it's because VxWorks doesn't have a way to wait for random pool initialization, but I don't know – this got added in #64372 without any discussion. I suppose it's a reasonable way to busy-wait for RNG initialization (which is what the `randSecure` call is all about).

Reply:
@joboet Ah, I see. Well, it's for tier 3 targets, so the maintainer of that target can decide for themselves if they want a better approach. *shrug*